gpu/gpu.py

```diff
@@ -0,0 +1,64 @@
+from random import random
+
+
+def check_squared_value(x, xSquer, testValue):
+    if x != testValue - 1:
+        return False
+    elif xSquer == 1:
+        return False
+    else:
+        return True
+
+
+# Run function on gpu "repetitions" cores
+def testing_value(testValue, testValueMinusOne, powerOfTwo):
+    # Choose a random number between 2 and testValue - 2
+    randomValue = int(random() * (testValue - 3)) + 2
+
+    # Calculate randomValue^testValueMinusOne mod testValue
+    x = pow(randomValue, testValueMinusOne, testValue)
+
+    # If x is 1 or testValue - 1, we don't know if testValue is prime, so we'll try again
+    if x == 1 or x == testValue - 1:
+        return True
+
+    # For "powerOfTwo" gpu cores:
+    for _ in range(1, powerOfTwo):
+
+        # Divide between gpu cores?
+        #
+        xSquer = pow(x, 2, testValue)
+        check_squared_value(x, xSquer, testValue)
+        x = xSquer
+        #
```

Comment:
Teraz tak patrzę to tej części chyba nie da się zrównoleglić jeśli kazda następna iteracja bazuje na poprzedniej. Chyba, ze jest jakaś technika na to ale nie sądzę. 
Dodatkowo to chyba zbyt proste operacje, które nie nadają się na zrównoleglenie gdy nie przyniesie to zysku w czasie wykonania. ale warto sprawdzić, w sumie na tym polega ten projekt xd

Reply:
Moze przekopiowac tą petlę 1:1 z CPU tu zeby nie zaburzyc logiki i podjąć tą decyzję jak będziemy mieć większą w tym temacie wiedzę, Co myślicie?

Reply:
Sam się właśnie nad tym zastanawiałem, dlatego trochę zrobiłem te funkcje na czuja...

Reply:
Możemy tak zrobić na razie i tak pewnie wyjdzie wszystko jak się zrobi już podział na wątki